t/lib/OpenQA/Test/Utils.pm

```diff
@@ -93,7 +93,23 @@ sub cache_minion_worker {
             require OpenQA::CacheService;
             local $ENV{MOJO_MODE} = 'test';
             note('Starting cache minion worker');
-            OpenQA::CacheService::run(qw(run --dequeue-timeout 1));
+            OpenQA::CacheService::run(
+                ['run', '--dequeue-timeout', '1'],
+                sub {
+                    my $app = shift;
+                    $app->minion->on(
+                        worker => sub {
+                            my ($minion, $worker) = @_;
+                            $worker->on(
+                                dequeue => sub {
+                                    my ($worker, $job) = @_;
+                                    $job->on(
+                                        cleanup => sub {
+                                            Devel::Cover::report() if Devel::Cover->can('report');
+                                        });
+                                });
+                        });
+                });
```

Comment:
one has to admire such prettyness :)

Would this work though

```suggestion
            OpenQA::CacheService::run(
                ['run', '--dequeue-timeout', '1'],
                sub ($app) {
                    $app->minion->on(worker => sub ($minion, $worker) {
                            $worker->on(dequeue => sub ($worker, $job) {
                                    $job->on(cleanup => sub { Devel::Cover::report() if Devel::Cover->can('report') });
                                });
                        });
                });
```

Reply:
Sure, i don't want to covert the whole file to signatures though. 😉

Reply:
Oh, the file is already partially converted.